### DISCUSSION TITLE
AIS-catcher & RTL-SDR V4 support & unique sharing key (UUID) issue when running as a service

### DISCUSSION THREAD

**Discussion by DaveBMD** [jvde-github/AIS-catcher#303] [General]
First started AIS-catcher using the RTL-SDR V4 with no data being received, switched to and RTL-SDR V3 and everything worked fine. So apparently RTL-SDR V4 is not supported.

Also I'm running AIS-catcher as a service using config.cmd for the command line variables. When I add in the unique sharing key (UUID) to the -X line, restarting the service fails. Removing the unique sharing key (UUID) and leaving just the -X by itself, restarting the service all functions work.

**Comment by jvde-github** [comment]
Hello,

Can you tell me what version you are running in the service? E.g from the log printout.

Also, how did you install the V4 driver and do you see that the V4 blog is detected in the output?

**Reply by tkreyche** [reply]
Hi Jasper,

I just had the same experience with a brand new V4. I just swapped a V3 for the V4 on a working system, and it showed nothing received, not even from the strong base station near my house. Even the basic startup AIS-catcher -v 5 shows nothing, and also tried on a second newly built Pi/OS/AIS-catcher system.

The first AIS-catcher run is from the V4 (Rafael Micro R828D which receives nothing), the second from the V3 (Rafael Micro R820T) which works fine.

Any suggestions? 

thanks, Tom

Using the latest librtlsdr-dev driver and AIS-catcher build:
```
pi@raspberrypi:~ $ sudo apt install librtlsdr-dev
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
librtlsdr-dev is already the newest version (0.6.0-3).
0 upgraded, 0 newly installed, 0 to remove and 80 not upgraded.


pi@raspberrypi:~ $ AIS-catcher -gr tuner auto rtlagc on -a 192k -C config.json -v  60 -o 0 -X zzzzzzz
AIS-catcher (build Aug 13 2024) v0.60-16-g526aa2c6
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Rafael Micro R828D tuner
TCP feed: open socket for host: aiscatcher.org, port: 4242, filter: OFF, PERSIST: ON, KEEP_ALIVE: OFF, UUID: zzzzzzzz, JSON: ON, status: pending
UDP: open socket for host: 5.9.207.224, port: 5943, filter: OFF, JSON: OFF
UDP: open socket for host: 144.76.105.244, port: 2261, filter: OFF, JSON: OFF
UDP: open socket for host: 54.204.25.151, port: 7127, filter: OFF, JSON: OFF
UDP: open socket for host: 195.201.71.220, port: 6233, filter: OFF, JSON: OFF
UDP: open socket for host: hub.shipxplorer.com, port: 36081, filter: OFF, JSON: OFF
Server: reading statistics from stat.bin
TCP Server: start thread at port 8100
HTML Server running at port 8100
Server: starting backup service every 10 minutes.
[R82XX] PLL not locked!
[R82XX] PLL not locked!
[R82XX] PLL not locked!
Allocating 15 zero-copy buffers
Device    : Blog V4
Settings  : rate 1536K bw 192K format CU8 tuner AUTO rtlagc ON biastee OFF buffer_count 24
Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.60] ps_ema ON afc_wide ON droop ON fp_ds OFF dsk OFF
[AIS engine v0.60 #0-0]                 received: 0 msgs, total: 0 msgs, rate: 0 msg/s


pi@raspberrypi:~ $ AIS-catcher -gr tuner auto rtlagc on -a 192k -C config.json -v  60 -o 0 -X zzzzzzz
AIS-catcher (build Aug 13 2024) v0.60-16-g526aa2c6
(C) Copyright 2021-2024 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Rafael Micro R820T tuner
TCP feed: open socket for host: aiscatcher.org, port: 4242, filter: OFF, PERSIST: ON, KEEP_ALIVE: OFF, UUID: zzzzz JSON: ON, status: pending
UDP: open socket for host: 5.9.207.224, port: 5943, filter: OFF, JSON: OFF
UDP: open socket for host: 144.76.105.244, port: 2261, filter: OFF, JSON: OFF
UDP: open socket for host: 54.204.25.151, port: 7127, filter: OFF, JSON: OFF
UDP: open socket for host: 195.201.71.220, port: 6233, filter: OFF, JSON: OFF
UDP: open socket for host: hub.shipxplorer.com, port: 36081, filter: OFF, JSON: OFF
Server: reading statistics from stat.bin
TCP Server: start thread at port 8100
HTML Server running at port 8100
Server: starting backup service every 10 minutes.
Allocating 15 zero-copy buffers
Device    : RTL2838UHIDIR
Settings  : rate 1536K bw 192K format CU8 tuner AUTO rtlagc ON biastee OFF buffer_count 24
Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.60] ps_ema ON afc_wide ON droop ON fp_ds OFF dsk OFF
TCP (aiscatcher.org:4242): connected to server.
```

**Comment by jvde-github** [comment]
Hi Tom,

You have the latest drivers that come with the distributions but unfortunately these do not include the changes for the V4. In the documentation for the V4 (https://www.rtl-sdr.com/v4/) they mention you have to remove the system driver:

```
sudo apt purge ^librtlsdr
sudo rm -rvf /usr/lib/librtlsdr* /usr/include/rtl-sdr* /usr/local/lib/librtlsdr* /usr/local/include/rtl-sdr* /usr/local/include/rtl_* /usr/local/bin/rtl_*
````
You can install AIS-catcher via my script (which installs rtl-sdr from source if it is not on the system to avoid all these issues for fresh systems):
https://github.com/jvde-github/AIS-catcher?tab=readme-ov-file#installation-1

Or... install yourself:
```
    git clone https://gitea.osmocom.org/sdr/rtl-sdr --depth 1
    cd rtl-sdr
    mkdir build && cd build 
    cmake ../ -DINSTALL_UDEV_RULES=ON  -DDETACH_KERNEL_DRIVER=ON
    make 
    make install
    cp ../rtl-sdr.rules /etc/udev/rules.d/
    ldconfig 
    cd ../.. 
````
And then rebuild AIS-catcher as well so it links to the new system.

Apologies for that, but unless RTL-SDR sorts out their driver package it remains a bit cumbersome.

**Comment by jvde-github** [comment]
Note that the new install/update script installs in /usr/bin (old make files might have put it in /usr/local/bin so might have to delete that to start the newest).

Also best to remove the sharing keys in your original post (or I can send you a new one :-) )

**Comment by tkreyche** [comment]
Thanks Jasper, working now...deleted the old drivers and reinstalling using your script. Also removed my key....Tom
